pulsar/internal/topic_name.go

```diff
@@ -44,9 +44,10 @@ func ParseTopicName(topic string) (*TopicName, error) {
 	if !strings.Contains(topic, "://") {
 		// The short topic name can be:
 		// - <topic>
-		// - <property>/<namespace>/<topic>
+		// - <tenant>/<namespace>/<topic>
+		// - <tenant>/<cluster>/<namespace>/<topic>
 		parts := strings.Split(topic, "/")
-		if len(parts) == 3 {
+		if len(parts) == 3 || len(parts) == 4 {
```

Comment:
Overall looks good. Let's add a test case for this.

Reply:
```log
Running tool: /usr/local/bin/go test -timeout 30s github.com/apache/pulsar-client-go/pulsar/internal -run ^(TestParseTopicName)$

ok  	github.com/apache/pulsar-client-go/pulsar/internal	0.559s
Success: Tests passed.
```